// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mongoose model wrapper function adaptMongooseModel() - when looking at the code of single methods in production mode - overloads find() matching snapshot 1`] = `
"async function wrapper(...methodArgs) {
  let mongooseError = new Error(
    \`kth-node-cosmos-db: Internal error - Failed to call Mongoose method \\"\${methodName}\\"\`
  );
  let errorHandlerResult = {};
  const strategyName = client.getOption(\\"retryStrategy\\");
  const config = StrategyUtils.getRetryStrategy(strategyName);
  const { timeoutPerAttempt, maxAttempts, lastTimeout } = config;
  for (let attempt = 0; attempt < maxAttempts; attempt++) {
    const timeout =
      attempt < timeoutPerAttempt.length
        ? timeoutPerAttempt[attempt]
        : lastTimeout;
    await _wait(timeout);
    mongooseError = null;
    try {
      const methodData = {
        model,
        methodName,
        methodArgs,
        supportBatchsize,
        client
      };
      const result = await _callMongooseMethodAsync(methodData);
      return result;
    } catch (error) {
      mongooseError = error;
    }
    errorHandlerResult = {};
    try {
      const errorHandlerData = {
        model,
        error: mongooseError,
        client
      };
      errorHandlerResult = await _tryToHandleThroughputError(errorHandlerData);
    } catch (internalError) {
      log.error(
        \\"kth-node-cosmos-db: Increasing throughput during Mongoose operation failed\\",
        {
          mongooseError,
          internalError
        }
      );
      throw internalError;
    }
    if (!errorHandlerResult.isThroughputError) {
      throw mongooseError;
    }
    if (errorHandlerResult.throughputMaximal) {
      break;
    }
  }
  throw new Error(
    \`kth-node-cosmos-db: Failed during Mongoose method \\"\${methodName}\\" - Maybe there is generally too little throughput available\`
  );
}
"
`;
